src/dataflow/src/render/debezium.rs

```diff
@@ -118,6 +121,265 @@ where
+                                Some(ref mut s) => {
+                                    let res = s.should_use_record(
+                                        key,
+                                        &value,
+                                        result.position,
+                                        result.upstream_time_millis,
+                                        &debug_name,
+                                    );
+                                    match res {
+                                        Ok(b) => b,
+                                        Err(err) => {
+                                            // We could theoretically use tx_cap_map.get(tx_id) here but for sake of
+                                            // consistency, we output all errors at the data_cap time.
+                                            output.session(&data_cap).give((
+                                                Err(err),
+                                                *data_cap.time(),
+                                                1,
+                                            ));
+                                            continue;
+                                        }
+                                    }
+                                }
+                                None => true,
+                            };
+
+                            if should_use {
+                                let tx_cap_entry = match tx_cap_map.entry(tx_id.clone()) {
+                                    Entry::Occupied(e) => e,
+                                    Entry::Vacant(_) => panic!("Must have cap if using record"),
+                                };
+
+                                let mut session = output.session(tx_cap_entry.get());
+                                match value.iter().nth(before_idx).unwrap() {
+                                    Datum::List(l) => {
+                                        session.give((Ok(Row::pack(&l)), tx_time, -1));
+                                    }
+                                    Datum::Null => {}
+                                    d => {
+                                        panic!("type error: expected record, found {:?}", d)
+                                    }
+                                }
+                                match value.iter().nth(after_idx).unwrap() {
+                                    Datum::List(l) => {
+                                        session.give((Ok(Row::pack(&l)), tx_time, 1));
+                                    }
+                                    Datum::Null => {}
+                                    d => {
+                                        panic!("type error: expected record, found {:?}", d)
+                                    }
+                                }
+                                match tx_event_count.entry(tx_id.clone()) {
```

Comment:
same here, you can use `get_mut(&tx_id).expect(..)`

Reply:
(well, here you need `.unwrap_or_else(|| panic!(..))` but it's still simpler than doing through the entry API

Reply:
I was going for a similar thing here: it seemed to minimize panics unwraps when we only sometimes remove the entry after reading it.  Is there a magical rust way to do that if we use `get_mut`?

Reply:
You're right, I didn't read it carefully enough, sorry. I'll resolve both of these comments